node/src/types/block.rs

```diff
@@ -1324,4 +1341,26 @@ mod tests {
             unexpected => panic!("Bad check response: {:?}", unexpected),
         }
     }
+
+    #[test]
+    fn finality_signature() {
+        let mut rng = TestRng::from_seed([2u8; 16]);
+        let block = Block::random(&mut rng);
+        // Signature should be over both block hash and era id.
+        let (secret_key, public_key) = asymmetric_key::generate_ed25519_keypair();
+        let secret_rc = Rc::new(secret_key);
+        let era_id = EraId(1);
+        let fs = FinalitySignature::new(*block.hash(), era_id, &secret_rc, public_key, &mut rng);
+        assert!(fs.verify().is_ok());
+        let signature = fs.signature;
+        // Verify that signatures includes era id.
```

Comment:
```suggestion
        // Verify that signature includes era id.
```

Reply:
Done in [`c0bd13f` (#683)](https://github.com/CasperLabs/casper-node/pull/683/commits/c0bd13f70e41208382667535631dd62c9a381c5c)